File: Backend/notification/models/notification.py
from datetime import datetime

from sqlalchemy import Column, String, Boolean, DateTime, Text
import uuid
from config.db import Base

def generate_uuid():
    return str(uuid.uuid4())

class Notificacion(Base):
    __tablename__ = "notificaciones"
    
    id = Column(String, primary_key=True, default=generate_uuid)
    user_id = Column(String, nullable=False, index=True)
    tipo = Column(String, nullable=False) # 'confirmed', 'discount', 'checkin_reminder', 'profile_update'
    titulo = Column(String, nullable=False)
    cuerpo = Column(Text, nullable=False)
    reserva_id = Column(String, nullable=True)
    timestamp = Column(DateTime(timezone=True), default=datetime.utcnow)
    leida = Column(Boolean, default=False)

class DeviceToken(Base):
    __tablename__ = "device_tokens"
    
    id = Column(String, primary_key=True, default=generate_uuid)
    user_id = Column(String, nullable=False, unique=True, index=True)
    token = Column(String, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

File: Frontend/mobile-app/lib/models/notification.dart
class NotificationModel {
  final String id;
  final String tipo;
  final String titulo;
  final String cuerpo;
  final String? reservaId;
  final DateTime timestamp;
  final bool leida;

  NotificationModel({
    required this.id,
    required this.tipo,
    required this.titulo,
    required this.cuerpo,
    this.reservaId,
    required this.timestamp,
    required this.leida,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'] ?? '',
      tipo: json['tipo'] ?? '',
      titulo: json['titulo'] ?? '',
      cuerpo: json['cuerpo'] ?? '',
      reservaId: json['reserva_id'],
      timestamp: DateTime.parse(json['timestamp']).toLocal(),
      leida: json['leida'] ?? false,
    );
  }
}
